src/tools/pkcs11-tool.c

```diff
@@ -2750,6 +2794,27 @@ static void encrypt_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 		util_fatal("Mechanism %s illegal or not supported\n", p11_mechanism_to_name(opt_mechanism));
 	}
 
+	/* If an RSA-OAEP mechanism, it needs parameters */
+	if (oaep_params.hashAlg) {
+		if (opt_mgf != 0)
+			oaep_params.mgf = opt_mgf;
+
+		/* These settings are compatible with OpenSSL 1.0.2L and 1.1.0+ */
+		oaep_params.source = 0UL;	 /* empty encoding parameter (label) */
```

Comment:
You do not seem to handle the label/param argument of the OAEP encryption.

I think this needs to be `CKZ_DATA_SPECIFIED` in all cases though, at least thats how I read the following part of PKCS#11 specs:

https://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/os/pkcs11-curr-v3.0-os.html#_Toc30061136

Reply:
I factorized the code and set CKZ_DATA_SPECIFIED in all the cases.
I just pushed the changes but I need more time to validate them.
I will let you know when the validation will be OK.